src/Orleans/Utils/DictionaryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans
+{
+    static class DictionaryExtensions
+    {
+        public static TResult Find<TKey, TResult>(this IDictionary<TKey, TResult> dictionary, TKey key) where TResult : class
```

Comment:
By analogy of driving a car if one always sets the turn signal when required whilst driving a car, there's no need to decide if it should be used when manouvering. One just does it as part of normal routine. It sets a consistent tone for the codebase. One just comments and no extra judgement is needed if it should be there or not or if in some case if it is self-explanatory or not. 

I don't know how to develop this particular codebase, so I opened #209 to discuss commenting, amonst other things.